### PR TITLE
Use prometheus v2.10.0

### PR DESCRIPTION
**Pull request by sylr**
Signed-off-by: Sylvain Rabot <s.rabot@lectra.com>

### CODE REVIEW

**Comment by brancz** [comment]
Not against this at all, but just trying to understand why we're making this change. Is it just so we are actually testing against this version? (I'd be happy with that, just trying to understand)

**Comment by sylr** [comment]
I think the amount of changes needed in thanos if we keep track of prometheus is smaller and easier to validate.

If we let prometheus in the same version and decide to bump it to let's 5~10 versions then the amount of work to incorporate the new version grows consequently and is prone to mistake.

File: pkg/util/labels.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package util
```
Comment (sylr) [inline pkg/util/labels.go]: pkg/util/labels.go what maybe not the good choice.

What about a new pkg/utils tree where we would move all xxxutil packages, ie.:

- pkg/utils/labelutil/*.go
- pkg/utils/runutil/*.go
- pkg/utils/strutil/*.go
...

What do you think @bwplotka ?

**Comment by brancz** [comment]
> If we let prometheus in the same version and decide to bump it to let's 5~10 versions then the amount of work to incorporate the new version grows consequently and is prone to mistakes.

Perfectly happy with that, and totally agree!

File: pkg/util/labels.go (hunk 1 above)
Comment (brancz) [reply pkg/util/labels.go]: just flattening this to 

* pkg/labelutil/*.go
* pkg/runutil/*.go
* pkg/strutil/*.go

sounds good to me as well.

**Comment by bwplotka** [comment]
Yea, testing is not the case here, only to use latest fixes and be up to date. Fully agree with @sylr that it's painful and more riskt to do it rarely.

Thanks for doing it!

**Pull request closed without merging**